types/ember-data/types/registries/model.d.ts

```diff
@@ -2,5 +2,7 @@
  * Catch-all for ember-data.
  */
 export default interface ModelRegistry {
+  // Allow any for the sake of catch-all
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
   [key: string]: any;
```

Comment:
It seems most of our models are missing the ModelRegistry code: https://docs.ember-cli-typescript.com/ts/using-ts-effectively#ember-data-lookups I wonder if this catch-all is needed if we add those. Or maybe it can be set to the base `Model` class?

Reply:
I've tried Model, but it gave a bunch of other errors. I'll try to implement the ModelRegistery!

Reply:
Yea, I don't think Model will work since everything is a subclass and adds extra fields. But I don't like the `any` since that is how it went unnoticed that the model registry entries were missing 😄.

So I would indeed add the registries (doesn't have to be this PR) and then we can maybe change this to `Model` (or `unknown`?) so that code fails if the registry entry is missing.

Reply:
I'll indeeed move it it to its own Issue so this PR can be finished!